Allow to turn off the sicky mode by setting v-sticky="null"

0m4r opened the pull request:
Allow to turn off the sticky mode by setting v-sticky="null" and to
restore the sticky behaviour by setting a new configuration object.
Example page updated to show the feature

Commit:
Allow to turn off the sicky mode by setting v-sticky="null"

Allow to turn off the sicky mode by setting v-sticky="null" and to
restore the sticky behaviour by setting a new configuration object.
Example page updated to show the feature

0m4r commented:
@rguanghui any chance you have a look at this pull request?

iqingting commented on src/index.js:
```diff
@@ -20,8 +25,14 @@ export default {
+    } else {
+      if (window.console && window.console.warn) {
+        window.console.warn('el.firstElementChild is undefined or null', el)
+      }
```
it's unnecessary to judgment window.console

0m4r replied:
not sure I get this, you say it is unnecessary to check for the availability of ```window.console``` or to output the log itself?

iqingting commented on src/index.js:
```diff
@@ -1,14 +1,19 @@
 let stickyTop
 let zIndex
+let className
+let childStyle
```
define under the line to use is better

iqingting commented on src/index.js:
```diff
@@ -1,14 +1,19 @@
     stickyTop = params.stickyTop || 0
     zIndex = params.zIndex || 1000
-
+    className = params.className || 'vue-sticky'
```
it's unnecessary, you can use like

`<div v-sticky class="xx"></div>`

iqingting commented on src/index.js:
```diff
@@ -32,21 +43,26 @@ export default {
     const check = () => {
       const offsetTop = el.getBoundingClientRect().top
-      if (offsetTop <= stickyTop) {
+      if (stickyTop !== null && offsetTop <= stickyTop) {
```
`stickyTop = params.stickyTop || 0`

so this cannot be `null`

0m4r replied:
well, with the changes I have made you could set the ```v-sticky=null``` to turn it off so, ```stickyTop``` could actually be ```null```

Commit:
Allow to turn off the sticky mode

Allow to turn off the sticky mode by setting a 'disabled' property to
the configuration object. Setting it to 'false' disable the sticky
behaviour. If set to 'true' or not set at all it enables it.

A class is added to the elment when in 'sticky' position and it is
removed when it is in its 'normal' position. The classe needs to
bespecified adding a propety 'className' to the configuration object.

Code has been improved to handle more than one sticky element present
in the same _page_.

Example app has been updated to showcase the code changes.

Commit:
Update README.md file to include 'disabled'

Update README.md file to include 'disabled' property to turn off
and on the sticky behaviour.

rguanghui commented:
3.3.4 published
